include/kb_client.hpp

```diff
@@ -543,26 +602,18 @@ class Client {
     }
 
     /*
-     * Add formatted output of an Object to the stringstream.
+     * Add formatted output to a stringstream.
      */
-    void prettyStream(const std::pair<std::string, Object>& v, std::stringstream& ss) {
-        ss << v.first << ", " << v.second.dtype();
-
-        if (v.second.dtype() == "MSGPACK_OBJECT_ARRAY") { // msgpack::ARRAY
-            ss << ", " << v.second.subType() << ", [" << v.second.size() << "]\n";
-
-        } else if (v.second.dtype() == "MSGPACK_OBJECT_MAP") { // msgpack::MAP
-            ss << " (Check...unexpected data type!)\n";
-
-        } else if (v.second.dtype() == "MSGPACK_OBJECT_BIN") { // msgpack::BIN
-            ss << ", " << v.second.subType() << ", [" << v.second.size() << "]\n";
+    template <typename T>
+    void prettyStream(const std::pair<std::string, T>& v, std::stringstream& ss) {
+        ss << v.first
+           << ", " << v.second.containerType()
+           << ", " << vectorToString(v.second.shape())
+           << ", " << v.second.dtype();
 
-        } else if (v.second.dtype() == "MSGPACK_OBJECT_EXT") { // msgpack::EXT
-            ss << " (Check...unexpected data type!)\n";
-
-        } else {
-            ss << "\n";
-        }
+        if (v.second.containerType() == "map" || v.second.containerType() == "MSGPACK_OBJECT_EXT")
+            ss << " (Check...unexpected data type!)";
```

Comment:
can we print more information on the output here?

Reply:
For the unknown data type, it will print something like
detector.data, map, [16], "undefined" (Check...unexpected data type!)

what else do you think we need?